File: WebAPI.Domain/Entities/Cadastro.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;

namespace APICadastro.Domain.Entities
{
    public class Cadastro
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string Nome { get; set; }
        public DateTime? DataNascimeto { get; set; }
        public string Facebook { get; set; }
        public string Linkedin { get; set; }
        public string Twitter { get; set; }
        public string Instagram { get; set; }
        public string CPF { get; set; }
        public string RG { get; set; }
        public List<Telefone> Telefones { get; set; }
        public List<Endereco> Enderecos { get; set; }
    }
}

File: AppCadastro/src/app/models/Cadastro.ts
export class Cadastro {
  id: string;
  nome: string;
  dataNascimeto?: Date;
  facebook: string;
  linkedin: string;
  twitter: string;
  instagram: string;
  cpf: string;
  rg: string;
  enderecos: Endereco[];
  telefones: Telefone[];
}

export class Endereco {
  identificacao: string;
  logradouro: string;
  cidade: string;
  uf: string;
}

export class Telefone {
  identificacao: string;
  dDD: string;
  numeroTelefone: string;
}

export class Estado {
  uf: string;
  nome: string;
}

